### PR TITLE
Display nodeName of connectorPod in e2e tests

### PR DESCRIPTION
**Pull request by sridhargaddam**
While looking at the e2e logs, it was seen that sometimes the node on which the connectorPod was scheduled is not displayed. This is an important information which will be helpful while debugging any failures. This PR fixes it.

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Comment by sridhargaddam** [comment]
Sample logs from one of the CI jobs: 

In Line2 below, we can see that nodeName is empty for the connector pod.

```
2022-10-14T11:46:04.1361004Z [1mSTEP[0m: Creating a connector pod in cluster "cluster1", which will attempt the specific UUID handshake over TCP
2022-10-14T11:46:04.1361611Z [1mSTEP[0m: Waiting for the connector pod "tcp-check-podc69qn" on node "" to exit, returning what connector sent
2022-10-14T11:46:04.1362088Z Oct 14 11:45:14.123: INFO: Pod "tcp-check-podc69qn" output:
2022-10-14T11:46:04.1362364Z nc: 10.131.0.9 (10.131.0.9:1234): Host is unreachable
2022-10-14T11:46:04.1362628Z nc: 10.131.0.9 (10.131.0.9:1234): Host is unreachable
2022-10-14T11:46:04.1362778Z 
2022-10-14T11:46:04.1363192Z [1mSTEP[0m: Waiting for the listener pod "tcp-check-listenergmk94" on node "cluster2-control-plane" to exit, returning what listener sent
2022-10-14T11:46:04.1363721Z Oct 14 11:46:04.128: INFO: Pod "tcp-check-listenergmk94" output:
2022-10-14T11:46:04.1363983Z listening on 0.0.0.0:1234 ...
2022-10-14T11:46:04.1364195Z nc: timeout
```

**Comment by sridhargaddam** [comment]
Sample output with the fix from this PR:

```
STEP: Creating a connector pod in cluster "cluster2", which will attempt the specific UUID handshake over TCP
STEP: Waiting for the connector pod "tcp-check-podcgntg" to exit, returning what connector sent
Oct 14 18:42:50.553: INFO: Pod "tcp-check-podcgntg" on node "cluster2-worker2" output:
100.66.15.18 (100.66.15.18:1234) open
[dataplane] listener says d323607e-f8a0-47ba-97c8-76a1b39c7fd9

STEP: Waiting for the listener pod "tcp-check-listenerp8q6f" to exit, returning what listener sent
Oct 14 18:42:50.561: INFO: Pod "tcp-check-listenerp8q6f" on node "cluster1-worker" output:
listening on 0.0.0.0:1234 ...
connect to 10.130.1.17:1234 from 10.131.1.13:42299 (10.131.1.13:42299)
[dataplane] connector says b54ea875-e72f-432a-9307-d4fa976df5b9
```

File: test/e2e/tcp/connectivity.go [hunk 1]
```diff
@@ -174,12 +174,10 @@ func createPods(p *ConnectivityTestParams) (*framework.NetworkPod, *framework.Ne
 		framework.Logf("Will send traffic from IP: %v", sourceIP)
 	}
 
-	By(fmt.Sprintf("Waiting for the connector pod %q on node %q to exit, returning what connector sent",
-		connectorPod.Pod.Name, connectorPod.Pod.Spec.NodeName))
+	By(fmt.Sprintf("Waiting for the connector pod %q to exit, returning what connector sent", connectorPod.Pod.Name))
```
Comment (tpantelis) [inline test/e2e/tcp/connectivity.go:177]: Why did you remove printing the node name here? Isn't that contrary to the purpose of this PR?

Comment (sridhargaddam) [reply test/e2e/tcp/connectivity.go:177]: Cos' it will be duplicated as its getting printed as part of connectorPod.AwaitFinish() api below. 

Please see the sample output I pasted in the PR:
https://github.com/submariner-io/shipyard/pull/980#issuecomment-1279352310

**Review by dfarrell07 — APPROVED** [review]

File: test/e2e/tcp/connectivity.go (hunk 1 above)
Comment (tpantelis) [reply test/e2e/tcp/connectivity.go:177]: Right but that's only if `verbose` is enabled, which it is for CI but I assume not for `subctl`. I don't think it hurts to duplicate it anyway - we print the pod name in  both places so it would be consistent.

Comment (sridhargaddam) [reply test/e2e/tcp/connectivity.go:177]: If we do not pass "--verbose" as part of `subctl verify ...` command, we do not get any info including these log messages right? 
See sample output: https://pastebin.com/raw/yPg8Sd4q

Comment (tpantelis) [reply test/e2e/tcp/connectivity.go:177]: OK right. Still IMO it doesn't hurt to print it anyway - as I said we duplicate printing the pod name so why not the node name as well since we did before.  If we want to be consistent then we shouldn't duplicate the pod name either. Anyway if you feel strongly about it then I'm fine either way.

**Review by tpantelis — APPROVED** [review]

File: test/e2e/tcp/connectivity.go (hunk 1 above)
Comment (sridhargaddam) [reply test/e2e/tcp/connectivity.go:177]: The main issue is that sometimes its empty at this stage and could be misleading. 

```
Waiting for the connector pod "tcp-check-podc69qn" on node "" to exit, returning what connector sent
INFO: Pod "tcp-check-podc69qn" output:
```

The latest pod info gets updated as part of connectorPod.AwaitFinish() API where we have the node details associated with the pod.

**Comment by sridhargaddam** [comment]
Backport PR (release-0.13 branch): https://github.com/submariner-io/shipyard/pull/1022
